scripts/ci-entrypoint.sh

```diff
@@ -305,9 +303,10 @@ on_exit() {
     unset KUBECONFIG
     go run -tags e2e "${REPO_ROOT}"/test/logger.go --name "${CLUSTER_NAME}" --namespace default
     "${REPO_ROOT}/hack/log/redact.sh" || true
-    # cleanup
-    if [[ -z "${SKIP_CLEANUP:-}" ]]; then
-        cleanup
+    # cleanup all resources we use
+    if [[ ! "${SKIP_CLEANUP:-}" == "true" ]]; then
```

Comment:
why make the skip condition more strict? If the dev defined `SKIP_CLEANUP` I think it's safe to assume they want to skip, unless maybe it's false

Reply:
It's that latter thing. The obvious choices seem to be:

- Make `true` explicitly meaningful
- Make `false` and "" (empty string) and null meaningful.

The former is (IMO) easier to manage. If someone is passing in `SKIP_CLEANUP=foo` and assuming that's part of the contract, I think it's O.K. to regress that existing usage (because it doesn't make sense). On the other hand, for anyone passing in`SKIP_CLEANUP=false`, we may be finally doing the right thing.

Open to other thoughts here, but I think the `-z` is definitely the wrong thing here, as (IME) there is no obvious way to create a "boolean env var" in bash in the way that `-z` would make sense.

Reply:
fair, my opinion is loosely held